vsg/tests/vsg/test_vsg.py

```diff
@@ -481,3 +481,44 @@ def test_summary_output_format_multiple_mixed_jobs_2(self):
 
         self.assertEqual(utils.replace_total_count_summary(lActualStdErr), lExpectedStdErr)
         self.assertEqual(utils.replace_total_count_summary(lActualStdOut), lExpectedStdOut)
+
+    def test_globbing_filenames_in_configuration_with_file_rules(self):
+        lExpected = []
+        lExpected.append('ERROR: vsg/tests/vsg/entity2.vhd(8)port_008 -- Change number of spaces after *out* to 3.')
+        lExpected.append('')
+
+        try:
+            subprocess.check_output(['bin/vsg','--configuration','vsg/tests/vsg/config_glob_with_file_rules.yaml','--output_format','syntastic'])
+        except subprocess.CalledProcessError as e:
+            lActual = str(e.output.decode('utf-8')).split('\n')
+            iExitStatus = e.returncode
+
+        self.assertEqual(lActual, lExpected)
+
+    def test_configuration_with_file_rules_and_no_file_list_entity2(self):
+        lExpected = []
+        lExpected.append('ERROR: vsg/tests/vsg/entity2.vhd(8)port_008 -- Change number of spaces after *out* to 3.')
+        lExpected.append('')
+
+        try:
+            subprocess.check_output(['bin/vsg','--configuration','vsg/tests/vsg/config_file_rules.yaml', '-f', 'vsg/tests/vsg/entity2.vhd', '--output_format','syntastic'])
+        except subprocess.CalledProcessError as e:
+            lActual = str(e.output.decode('utf-8')).split('\n')
+            iExitStatus = e.returncode
+
+        self.assertEqual(lActual, lExpected)
+
+    def test_configuration_with_file_rules_and_no_file_list_entity1(self):
+        lExpected = []
+
+        lActual = []
+
+        try:
+            subprocess.check_output(['bin/vsg','--configuration','vsg/tests/vsg/config_file_rules.yaml', '-f', 'vsg/tests/vsg/entity1.vhd', '--output_format','syntastic'])
+        except subprocess.CalledProcessError as e:
+            lActual = str(e.output.decode('utf-8')).split('\n')
+            iExitStatus = e.returncode
+
+        self.assertEqual(lActual, lExpected)
+
+
```

Comment:
@jeremiah-c-leary works nicely!

these trailing lines are on purpose?

Reply:
@alonbl ,

Those trailing lines are not on purpose.  I have committed an update to remove them.

I also refactored `configure_rules` to reduce the complexity, and hopefully make it easier to read.

--Jeremy

Reply:
Thank you!

The refactor is subjective :)
For me, splitting the code into so many functions for a single task is less easier to read, but that's not important.
Please review my other patches ;)